Docs/API_design_doc.md

```diff
@@ -90,3 +90,113 @@
+
+    * Request Method：`POST` or `GET`
+
+    * Request Format：`JSON`
+
+    * Response Format：`JSON`
+
+    * Note：Get scenario search result based on scenario of command keyword
+
+    * Request Parameters：
+
+        | Name    | Type          | Is Require | Default value | Description                                          | Supported |
+        | ------- | ------------- | ---------- | ------------- | ---------------------------------------------------- | --------- |
+        | keyword | string        | true       | -             | The search keyword user searches                     | Yes       |
+        | type    | int or string | false      | all           | Search type, value range: 1.all 2.scenario 3.command | Yes       |
+        | top_num | int           | false      | 5             | The maximum number of search results                 | Yes       |
+
+    * Response Data:
+
+        | Name   | Type        | Description                     |
+        | ------ | ----------- | ------------------------------- |
+        | status | int         | Status code                     |
+        | error  | JSON        | Error information               |
+        | data   | JSON (list) | [Search Result](#search_result) |
+
+        <span id = "search_result">Search Result</span>
+        | Name               | Type       | Description                               |
+        | ------------------ | ---------- | ----------------------------------------- |
+        | id                 | string     | Recommended command                       |
+        | source             | int        | Scenario source                           |
+        | commandSet         | json(list) | command sequence in scenario              |
+        | scenario           | string     | scnario name                              |
+        | firstCommand       | string     | first command in scenario                 |
+        | source_url         | string     | link to origin file                       |
+        | update_time        | string     | when the scenario updated                 |
+        | description        | string     | scenario description                      |
+        | score      | float      | Search score                              |
+        | highlights | json       | highlight related content with &lt;em&gt; |
+
+    * Example：
+        
+        1. Search Scenario Example:
+        Request:
+        ```http
+                POST http://localhost:7071/api/SearchService HTTP/1.1
+                Content-Type: application/json
+
+                {
+                    "keyword": "scale+server",
+                    "top": 5,
+                    "type": "Scenario"
+                }
+        ```
+
+        Response
+        ```json
+        {
+            "data": [
+                {
+                    "id": "az postgres server create | scale postgresql server - Azure CLI Samples",
```

Comment:
It seems that the `id` is useless. Can we remove this redundant attribute in our logic (include Cosmos DB)?

Reply:
> It seems that the `id` is useless. Could we remove this redundant attribute in our logic (include Cosmos DB)?

We use a upsert logic which relies on unique identifiers when updating scenario data in Cosmos DB. We need an `id` property to make sure we can replace the correct document when the scenario is updatesd. 

But I agree that `id` should be removed from the service response.